libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/SpecTranslate/Conway.hs

```diff
@@ -0,0 +1,875 @@
+      <*> transHashProtected ensCurPParams grPParamUpdate
+      <*> transWithdrawals ensWithdrawals
+    where
+      GovRelation {..} = ensPrevGovActionIds
+      transWithdrawals ws = forM (Map.toList ws) $
+        \(cred, Coin amount) -> do
+          agdaCred <- toSpecRep cred
+          pure (((), agdaCred), amount)
+      transHashProtected x h = do
+        committee <- toSpecRep x
+        agdaLastId <- case h of
+          SJust lastId -> toSpecRep lastId
+          SNothing -> pure (0, 0)
+        pure (committee, agdaLastId)
+
+instance
+  ( SpecTranslate ctx (PParamsHKD Identity era)
+  , Inject ctx (EnactState era)
+  , EraPParams era
+  , SpecRep (PParamsHKD Identity era) ~ Agda.PParams
+  ) =>
+  SpecTranslate ctx (GovEnv era)
+  where
+  type SpecRep (GovEnv era) = Agda.GovEnv
+  type SpecTransContext (GovEnv era) = EnactState era
+
+  toSpecRep GovEnv {..} = do
+    enactState <- askTransCtx @(GovEnv era)
+    Agda.MkGovEnv
+      <$> toSpecRep geTxId
+      <*> toSpecRep geEpoch
+      <*> toSpecRep gePParams
+      <*> toSpecRep gePPolicy
+      <*> toSpecRep enactState
+
+instance SpecTranslate ctx (Voter era) where
+  type SpecRep (Voter era) = Agda.Voter
+
+  toSpecRep (CommitteeVoter c) = (Agda.CC,) <$> toSpecRep c
+  toSpecRep (DRepVoter c) = (Agda.DRep,) <$> toSpecRep c
+  toSpecRep (StakePoolVoter kh) = (Agda.SPO,) <$> toSpecRep (KeyHashObj kh)
+
+instance SpecTranslate ctx Vote where
+  type SpecRep Vote = Agda.Vote
+
+  toSpecRep VoteYes = pure Agda.VoteYes
+  toSpecRep VoteNo = pure Agda.VoteNo
+  toSpecRep Abstain = pure Agda.VoteAbstain
+
+instance SpecTranslate ctx (VotingProcedures era) where
+  type SpecRep (VotingProcedures era) = [Agda.GovSignal]
+
+  toSpecRep = foldrVotingProcedures go (pure [])
+    where
+      go ::
+        Voter (EraCrypto era) ->
+        GovActionId (EraCrypto era) ->
+        VotingProcedure era ->
+        SpecTransM ctx [Agda.GovSignal] ->
+        SpecTransM ctx [Agda.GovSignal]
+      go voter gaId votingProcedure m =
+        (:)
+          <$> fmap
+            Agda.GovSignalVote
+            ( Agda.MkGovVote
+                <$> toSpecRep gaId
+                <*> toSpecRep voter
+                <*> toSpecRep (vProcVote votingProcedure)
+                <*> toSpecRep (vProcAnchor votingProcedure)
+            )
+          <*> m
+
+instance SpecTranslate ctx (GovAction era) where
+  type SpecRep (GovAction era) = Agda.GovAction
+
+  toSpecRep ParameterChange {} = pure $ Agda.ChangePParams ()
+  toSpecRep (HardForkInitiation _ pv) = Agda.TriggerHF <$> toSpecRep pv
+  toSpecRep (TreasuryWithdrawals withdrawals _) =
+    Agda.TreasuryWdrl
+      <$> toSpecRep withdrawals
+  toSpecRep (NoConfidence _) = pure Agda.NoConfidence
+  toSpecRep (UpdateCommittee _ remove add threshold) =
+    Agda.NewCommittee
+      <$> toSpecRep add
+      <*> toSpecRep remove
+      <*> toSpecRep threshold
+  toSpecRep (NewConstitution _ (Constitution anchor policy)) =
```

Comment:
How do we handle the hash protection on the Agda side if the previous governance action id is not translated to SpecRep?